integrations/Pendo/browser.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-underscore-dangle */
+import logger from "../../utils/logUtil";
+import ScriptLoader from "../ScriptLoader";
+
+class Pendo {
+  constructor(config, analytics) {
+    this.analytics = analytics;
+    this.apiKey = !config.apiKey ? "" : config.apiKey;
+    this.isDebugMode = config.enableDebugMode;
+    this.name = "PENDO";
+    this.setIntervalHandler = undefined;
+    logger.debug("Config ", config);
+  }
+
+  init() {
+    window.pendo = {};
+    window.pendoCli = undefined;
+    window.pendo.apiKey = this.apiKey;
+    ScriptLoader(
+      "pendo-fs",
+      `https://cdn.pendo.io/agent/static/${this.apiKey}/pendo.js`
+    );
+    logger.debug("===in init Pendo===");
+
+    // Make a call to initPendo until pendoCli gets references to pendo object.
+    this.setIntervalHandler = setInterval(this.initPendo.bind(this), 1000);
+  }
+
+  initPendo() {
+    if (this.isReady()) {
+      window.pendoCli = window.pendo;
+      clearInterval(this.setIntervalHandler);
+
+      // After pendo is initialized set debugger if debugMode is enabled.
+      if (this.isDebugMode) {
+        this.enableDebugging();
+      } else {
+        this.disableDebugging();
+      }
+    }
+  }
+
+  isLoaded() {
+    logger.debug("in PENDO isLoaded");
+    return !!window.pendoCli;
+  }
+
+  isReady() {
+    return window.pendo && window.pendo.isReady && window.pendo.isReady();
+  }
+
+  constructPendoAnonymousId(id) {
+    return `_PENDO_T_${id}`;
+  }
+
+  isUserAnonymous(message) {
+    return !message.userId;
+  }
+
+  identify(rudderElement) {
+    let visitorObj = {};
+    const id = this.isUserAnonymous(rudderElement.message)
+      ? this.constructPendoAnonymousId(rudderElement.message.anonymousId)
+      : rudderElement.message.userId;
+    visitorObj.id = id;
+    if (rudderElement.message.context.traits) {
```

Comment:
let { traits } = rudderElement.message.context... 
before if so that we do not need to get the value twice and just use the variable.

Reply:
Fixed